management/server/sql_store.go

```diff
@@ -311,6 +311,36 @@ func (s *SqlStore) SavePeerLocation(accountID string, peerWithLocation *nbpeer.P
 	return nil
 }
 
+// SaveUsers saves the given list of users to the database.
+// It updates existing users if a conflict occurs.
+func (s *SqlStore) SaveUsers(accountID string, users map[string]*User) error {
+	usersToSave := make([]User, 0, len(users))
+	for id, user := range users {
+		user.Id = id
+		user.AccountID = accountID
+		for id, pat := range user.PATs {
+			pat.ID = id
+			user.PATsG = append(user.PATsG, *pat)
+		}
+		usersToSave = append(usersToSave, *user)
+	}
+	return s.db.Session(&gorm.Session{FullSaveAssociations: true}).
+		Clauses(clause.OnConflict{UpdateAll: true}).
+		Create(&usersToSave).Error
+}
+
+// SaveGroups saves the given list of groups to the database.
+// It updates existing groups if a conflict occurs.
+func (s *SqlStore) SaveGroups(accountID string, groups map[string]*nbgroup.Group) error {
+	groupsToSave := make([]nbgroup.Group, 0, len(groups))
+	for id, group := range groups {
+		group.ID = id
```

Comment:
Do we really need to do this operation? In [this](https://github.com/netbirdio/netbird/pull/2245/files#diff-ad08fa670a36c3eed29a084065544822057a761069e7e4b94dd1613697bdf399R149) line I think we fill well the variable.

Reply:
Not necessarily needed, but it was to make sure the group ID is the same as the index